### PR TITLE
SimplePerceptron in P5

### PR DESCRIPTION
**Pull request by tonyslab**
My first contribution to the coding train, I saw your SimplePerceptron video and as i noticed its not available in JS/P5 I thought that I would try to make it :D.

### CODE REVIEW

**Comment by gruselhaus** [comment]
Hey @tonyslab,

thank you for this port. @shiffman ready to merge?

File: CodingChallenges/SimplePerceptron/P5/SimplePerceptron/index.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>Perceptron</title>
+    <script src="https://cdn.jsdelivr.net/npm/p5@1.1.9/lib/p5.min.js"></script>
+    <script src="Perceptron.js"></script>
```
Comment (shiffman) [inline CodingChallenges/SimplePerceptron/P5/SimplePerceptron/index.html:9]: small thing, but I prefer to use lower case filenames? (If that's silly or there's something I'm missing let me know!)

File: CodingChallenges/SimplePerceptron/P5/SimplePerceptron/Perceptron.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+// Machine Learning
+// Perceptron
+// More: http://natureofcode.com/book/chapter-10-neural-networks/
+// Video: https://youtu.be/ntKn5TPHHAk
+
+class Perceptron {
+  // Constructor
+  constructor() {
+    weights = [0, 0];
```
Comment (shiffman) [inline CodingChallenges/SimplePerceptron/P5/SimplePerceptron/Perceptron.js:13]:
```suggestion
    this.weights = [0, 0];
```

File: CodingChallenges/SimplePerceptron/P5/SimplePerceptron/Perceptron.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    lr = 0.1;
```
Comment (shiffman) [inline CodingChallenges/SimplePerceptron/P5/SimplePerceptron/Perceptron.js:14]:
```suggestion
    this.lr = 0.1;
```

File: CodingChallenges/SimplePerceptron/P5/SimplePerceptron/Perceptron.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    // Initialize the weights randomly
+    for (var i = 0; i < this.weights.length; i++) {
```
Comment (shiffman) [inline CodingChallenges/SimplePerceptron/P5/SimplePerceptron/Perceptron.js:16]: I prefer `let` to `var` these days!

**Review by shiffman — COMMENTED** [review]
Looks great, just a few changes suggested, thanks for the contribution!!

File: CodingChallenges/SimplePerceptron/P5/SimplePerceptron/index.html (hunk 1 above)
Comment (tonyslab) [reply CodingChallenges/SimplePerceptron/P5/SimplePerceptron/index.html]: Not sure why i did that tbh, thanks for the feedback!

**Pull request merged**